### PR TITLE
[ntsc-1.2] Import n64dd code from Decompollaborate/n64dd

### PR DESCRIPTION
**Pull request by cadmic**
Thanks for @AngheloAlf and @EllipticEllipsis for matching most of this code originally. There's not a great way to check that this matches (in fact it maybe doesn't since z_n64dd.c is bss-reordering-prone) but all the sections have the right size at least. In the n64dd repo it's matching for all N64 versions at least.

Most of this code is very undocumented but at least it's shiftable. Here's an overview:
* `z_n64dd.c`: Main entry point from the rest of OoT code. Starts background threads and provides functions to submit commands to them.
* `n64dd_data_buffer.c`: Contains a buffer that's only referenced in z_n64dd.c, but it's probably not part of that file because there's no way to match bss otherwise (it's referenced in z_n64dd .data and would always get reordered to the front). Potentially could be merged with another file
* `n64dd_801C8000.c`: Does some command processing in the background thread
* `n64dd_801C8940.c`: Lower-level command processing?
* `n64dd_801C9440.c`: Some text-handling routings
* `n64dd_801C9B70.c`: User-facing error handling
* `n64dd_error_headers.c`: Error message data. There's some .rodata padding so there's probably a file split in here somewhere 
* `n64dd_error_bodies.c`: More error message data. Probably could be merged with other files but not sure how to organize it
* `n64dd_error_textures.c`: Error message textures that are now extracted via ZAPD and stubbed out in non-N64 versions. Probably could be merged with other files but not sure how to organize it
* `n64dd_801CA0B0.c`: Draws text to the framebuffer

### CODE REVIEW

File: src/n64dd/n64dd_error_textures.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+#include "assets/n64dd/error_textures/n64dd_please_read_manual_jpn.i4.inc.c"
+    },
+};
+
+#else
+
+// Stub out missing assets
+u64 gN64DDError41Texs[2][0x600 / sizeof(u64)] = { 0 };
+u64 gN64DDPleaseReadManualTexs[2][0x2800 / sizeof(u64)] = { 0 };
```
Comment (Dragorn421) [inline src/n64dd/n64dd_error_textures.c:27]: ew, why?

File: src/n64dd/n64dd_801C9B70.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
+#include "global.h"
+#include "n64dd.h"
+
+u8 B_801E0F80[0x600];
+u8 B_801E1580[0x2800];
+u8 B_801E3D80[0x1400];
+
+/**
+ * @brief Converts a number in decimal to a hexadecimal number with the same digits, e.g. 1234 -> 0x1234.
```
Comment (Dragorn421) [inline src/n64dd/n64dd_801C9B70.c:9]:
```suggestion
 * Converts a number in decimal to a hexadecimal number with the same digits, e.g. 1234 -> 0x1234.
```

**Review by Dragorn421 — COMMENTED** [review]
extremely lenient review due to a combination of "kinda don't care for the n64dd" and "further passes may clean it up / document it"

File: src/n64dd/n64dd_error_textures.c (hunk 1 above)
Comment (cadmic) [reply src/n64dd/n64dd_error_textures.c:27]: For some reason I thought this was necessary to get it to compile. But on second thought it's not, we can just let it be a linker error

**Review by Dragorn421 — APPROVED** [review]

**Comment by AngheloAlf** [comment]
Could you add those file overviews you posted in the actual files?

**Comment by cadmic** [comment]
Alright, they're pretty low effort though because it's mainly vibes rather than a deep understanding of the code

**Comment by cadmic** [comment]
Well I think they're accurate, just intentionally rather vague

**Review by Dragorn421 — APPROVED** [review]

**Comment by cadmic** [comment]
Sneakily aligned n64dd segment to 0x40 here (thanks @AngheloAlf)

**Review by fig02 — APPROVED** [review]
ready to merge after conflict resolved

**Pull request merged**